src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -360,12 +360,19 @@ export class KernelProcess implements IKernelProcess {
             // First part of argument is always the executable.
             const executable = this.launchKernelSpec.argv[0];
             traceInfo(`Launching Raw Kernel & not daemon ${this.launchKernelSpec.display_name} # ${executable}`);
+
+            const useProcessEnv =
+                this._kernelConnectionMetadata.kind === 'startUsingKernelSpec' &&
+                this._kernelConnectionMetadata.useProcessEnv;
+
             const [executionService, env] = await Promise.all([
                 this.processExecutionFactory.create(this.resource),
                 // Pass undefined for the interpreter here as we are not explicitly launching with a Python Environment
                 // Note that there might still be python env vars to merge from the kernel spec in the case of something like
                 // a Java kernel registered in a conda environment
-                this.kernelEnvVarsService.getEnvironmentVariables(this.resource, undefined, this.launchKernelSpec)
+                !useProcessEnv
```

Comment:
Looking at this now...I think I could possibly just pack the process.env into the fake kernelspec. Not fully sure if I want the merge though.

Reply:
Can we flip this instead of looking for `!useProcessEnv`, why not look for `useProcessEnv ?...` Else its a mental switch

Reply:
Will this approach work for Conda environments? I guess the hypothesis is that users will have actiavted their conda environments isn't it?
My concern is users won't have done that and things will fall over.

Can we instead check if this is a conda environment & see if we have the conda env variable setup in process.env (then we'd know that the user hasn't activated the conda environment before launching VS Code).
One simple hack is to look for the name `conda` or the like in the python executable path (this way we are still fast and don't have a dependency on python extension)